include/deal.II/fe/mapping_fe.h

```diff
@@ -604,6 +607,18 @@ MappingFE<dim, spacedim>::preserves_vertex_locations() const
   return true;
 }
 
+
+template <int dim, int spacedim>
+bool
+MappingFE<dim, spacedim>::is_compatible_with(
+  const ReferenceCell::Type &cell_type) const
+{
+  if (cell_type.get_dimension() != dim)
+    return false; // TODO: or is this an error?
```

Comment:
Yes, it's an error. I think it would have been useful to equip `ReferenceCell` with a `<dim>` template argument, in which case this would become a compiler error.

Reply:
I would probably just have packed that into an `Assert`.